src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java

```diff
@@ -0,0 +1,404 @@
+        ActivityStateMachine.State, ActivityStateMachine.ExplicitEvent, ActivityStateMachine> {
+
+  enum ExplicitEvent {
+    SCHEDULE,
+    CANCEL
+  }
+
+  enum State {
+    CREATED,
+    SCHEDULE_COMMAND_CREATED,
+    SCHEDULED_EVENT_RECORDED,
+    STARTED,
+    COMPLETED,
+    FAILED,
+    TIMED_OUT,
+    CANCELED,
+    SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED,
+    SCHEDULED_ACTIVITY_CANCEL_EVENT_RECORDED,
+    STARTED_ACTIVITY_CANCEL_COMMAND_CREATED,
+    STARTED_ACTIVITY_CANCEL_EVENT_RECORDED,
+    COMPLETED_CANCEL_REQUESTED,
+    FAILED_CANCEL_REQUESTED,
+    CANCELED_CANCEL_REQUESTED,
+  }
+
+  private static StateMachine<State, ExplicitEvent, ActivityStateMachine> newStateMachine() {
+    return StateMachine.<State, ExplicitEvent, ActivityStateMachine>newInstance(
+            "Activity",
+            State.CREATED,
+            State.COMPLETED,
+            State.FAILED,
+            State.TIMED_OUT,
+            State.CANCELED)
+        .add(
+            State.CREATED,
+            ExplicitEvent.SCHEDULE,
+            State.SCHEDULE_COMMAND_CREATED,
+            ActivityStateMachine::createScheduleActivityTaskCommand)
+        .add(
+            State.SCHEDULE_COMMAND_CREATED,
+            CommandType.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK,
+            State.SCHEDULE_COMMAND_CREATED)
+        .add(
+            State.SCHEDULE_COMMAND_CREATED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED,
+            State.SCHEDULED_EVENT_RECORDED,
+            ActivityStateMachine::setInitialCommandEventId)
+        .add(
+            State.SCHEDULE_COMMAND_CREATED,
+            ExplicitEvent.CANCEL,
+            State.CANCELED,
+            ActivityStateMachine::cancelCommandNotifyCancelled)
+        .add(
+            State.SCHEDULED_EVENT_RECORDED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_STARTED,
+            State.STARTED,
+            ActivityStateMachine::setStartedCommandEventId)
+        .add(
+            State.SCHEDULED_EVENT_RECORDED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT,
+            State.TIMED_OUT,
+            ActivityStateMachine::notifyTimedOut)
+        .add(
+            State.STARTED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_COMPLETED,
+            State.COMPLETED,
+            ActivityStateMachine::notifyCompleted)
+        .add(
+            State.STARTED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_FAILED,
+            State.FAILED,
+            ActivityStateMachine::notifyFailed)
+        .add(
+            State.STARTED,
+            EventType.EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT,
+            State.TIMED_OUT,
+            ActivityStateMachine::notifyTimedOut)
+        .add(
+            State.SCHEDULED_EVENT_RECORDED,
+            ExplicitEvent.CANCEL,
+            State.SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED,
+            ActivityStateMachine::createRequestCancelActivityTaskCommand)
+        .add(
+            State.SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED,
```

Comment:
[Nit] swap the order of these two state transitions just for consistency

Reply:
Done